### PR TITLE
Saas 16.4 web fix selectmenu bottom area luvi

### PR DESCRIPTION
**Pull request by Iucapad**
[FIX] web: SelectMenu issues

The 1st commit fixes the style applied to the o_select_menu_sticky elements. Since the
fix from commit (1), the text cursor is shown when hovering those elements. This was
without considering the fact that this class is mainly used to display elements on
top of other elements of the component.
This includes usages with the bottomArea slot, that can be used with a DropdownItem.
Because of the changes from the commit previously named, the focused color was no longer
applied, and it was showing the wrong cursor when hovering.

The 2nd commit fixes an issue caused by the way we use onInput props in Knowledge to
fetch articles depending the current search value. Since we start from an empty
string in the input again, it make no sense to display the choices previously
fetched with a search value no longer displayed in the UI. The only way to display
the correct items is to update the input value, then put an empty search to fetch
accordingly.

This is clearly an issue, when the onInput props is used to fetch the content of
the SelectMenu, depending of the search value. The UI obviously display an empty
search, and filter accordingly, but without having called the fetching of the items
corresponding to the empty search value.

A test was added as well to assert SelectMenu can be used for this purpose without
forgetting to call onInput again when clearing the search value.

### CODE REVIEW

File: addons/web/static/tests/core/select_menu_tests.js [hunk 1]
```diff
@@ -1165,4 +1165,42 @@ QUnit.module("Web Components", (hooks) => {
...
+            />
+        `;
+
+        await mount(Parent, target, { env });
+        await open();
+        assert.verifySteps([], "onInput props has not been called initially");
+        await editInput(target, "input.o_select_menu_sticky", "a");
+        assert.verifySteps(["search=a"], "onInput props has been called with the right search string");
+
+        // closing the menu should clear the search string, trigger onInput and update the props
```
Comment (BastienFafchamps) [inline addons/web/static/tests/core/select_menu_tests.js:1200]: Shouldn't it be *opening now ?

File: addons/web/static/src/core/select_menu/select_menu.js [hunk 2]
```diff
@@ -139,17 +139,25 @@ export class SelectMenu extends Component {
...
-        this.isOpen = open;
-
-        if (!open) {
+    async onBeforeOpen() {
+        if (this.state.searchValue.length) {
             this.state.searchValue = "";
-            return;
+            if (this.props.onInput) {
+                // This props can be used by the parent to fetch items dynamically depending
+                // the search value. It must be called with with the empty search value.
```
Comment (BastienFafchamps) [inline addons/web/static/src/core/select_menu/select_menu.js:147]: "with with" :upside_down_face:

**Comment by BastienFafchamps** [comment]
@robodoo r+

**Comment by Iucapad** [comment]
@robodoo  rebase-ff

**Comment by BastienFafchamps** [comment]
@robodoo rebase-ff

**Comment by BastienFafchamps** [comment]
@robodoo r+

**Pull request closed without merging**